Public/Src/Engine/Processes/JobObject.cs

```diff
@@ -321,7 +321,7 @@ private bool TryGetProcessIds(uint length, out uint[] processIds, out uint requi
                     return false;
                 }
 
-                Contract.Assume(bytesWritten <= bufferSizeInBytes);
+                Contract.Assume(bytesWritten <= bufferSizeInBytes, $"More bytes written than the buffer size: {bytesWritten} > {bufferSizeInBytes}.");

```

Comment:
Can you put it inside an if-clause to avoid creating interpolated string?

```
if (bytesWritten > bufferSizeInBytes) 
{
    Contract.Assume(false, ...);
}
```